### PR TITLE
feat(argo-cd): Add support for extension

### PR DESCRIPTION
**Pull request by yq314**
This MR is intended to add support for ArgoCD extension, closes https://github.com/argoproj/argo-helm/issues/971

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/values.yaml
Comment (mbevc1) [inline charts/argo-cd/values.yaml:1280]:
```suggestion
  extensions: false
```

**Review by mbevc1 — COMMENTED** [review]
Do we also need to extend RBAC as well?
https://github.com/argoproj-labs/argocd-extensions/tree/main/manifests/rbac

File: charts/argo-cd/values.yaml
Comment (yq314) [reply charts/argo-cd/values.yaml:1280]: Ah yes, I missed this, will add

**Comment by yq314** [comment]
> Do we also need to extend RBAC as well? https://github.com/argoproj-labs/argocd-extensions/tree/main/manifests/rbac

Have added RBAC in the new commit, please take a look

File: charts/argo-cd/templates/argocd-server/deployment.yaml
Comment (mbevc1) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:141]: :thinking: if we want to parametrise this and pin it to a version. Not sure what upstream recommendation is TBH.

**Comment by mbevc1** [comment]
Thanks for the contribution, looks good. Small comment on container image label.

File: charts/argo-cd/templates/argocd-server/deployment.yaml
Comment (yq314) [reply charts/argo-cd/templates/argocd-server/deployment.yaml]: makes sense, I've extended the extensions section to allow configuring the images and also optionally load the rollouts-extension

File: charts/argo-cd/templates/argocd-server/extentions-role.yaml
Comment (mbevc1) [inline charts/argo-cd/templates/argocd-server/extentions-role.yaml:1]:
```suggestion
{{- if .Values.server.extensions.enabled }}
```

File: charts/argo-cd/templates/argocd-server/extensions-rolebinding.yaml
Comment (mbevc1) [inline charts/argo-cd/templates/argocd-server/extensions-rolebinding.yaml:1]:
```suggestion
{{- if .Values.server.extensions.enabled }}
```

File: charts/argo-cd/templates/argocd-server/deployment.yaml
Comment (mbevc1) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:179]:
```suggestion
      {{- if .Values.server.extensions.enabled }}
```

File: charts/argo-cd/templates/argocd-server/extentions-role.yaml
Comment (yq314) [reply charts/argo-cd/templates/argocd-server/extentions-role.yaml]: updated

File: charts/argo-cd/templates/argocd-server/deployment.yaml
Comment (yq314) [reply charts/argo-cd/templates/argocd-server/deployment.yaml]: updated

File: charts/argo-cd/templates/argocd-server/extensions-rolebinding.yaml
Comment (yq314) [reply charts/argo-cd/templates/argocd-server/extensions-rolebinding.yaml]: updated

**Review by mbevc1 — APPROVED** [review]
Looks good, thanks!

**Comment by mbevc1** [comment]
@mkilchhofer @davidkarlsen haven't managed to test this yet (Looks okay), but would appreciate second pair of eyes before merging ;)

File: charts/argo-cd/Chart.yaml
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:5]: Bump minor (as it adds a feature)
```suggestion
version: 3.28.0
```

**Review by mkilchhofer — COMMENTED** [review]
Beside the version bump I have another small point to discuss:
Would it make sense to mention that this is an early Argoproj subproject? (something like TechPreview or so?)

**Comment by yq314** [comment]
> Beside the version bump I have another small point to discuss: Would it make sense to mention that this is an early Argoproj subproject? (something like TechPreview or so?)

Hi Macro, thanks for the suggestion, I didn't know it is TechPreview actually. Where would be the best place to mention this? In the commend in the values file?

**Comment by yq314** [comment]
added a small comment to explain the techpreview status

**Review by mbevc1 — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**
